src/exchange.ts

```diff
@@ -29,10 +29,13 @@ interface DependentOperations {
 }
 
 // Returns the given operation with added __typename fields on its query
-const addTypeNames = (op: Operation): Operation => ({
-  ...op,
-  query: formatDocument(op.query),
-});
+const addTypeNames = (op: Operation): Operation => {
+  if (op.operationName === 'subscription') return op;
```

Comment:
so the problem here I suppose is that now we're never adding `__typename` to subscriptions? Will this need a patch on `urql`. I feel like we shouldn't ever add `__typename` to the toplevel maybe

Reply:
Well for mutation and query this is allowed since they allow batching. I don't really see the need for us to add the `__typename` on the top-level apart from that we now rely on it to do: `Query --> todos` and stuff. The most important thing about this fix is we can't alter the `query` on subscriptions